### PR TITLE
Moved Metadata to plugins. Also moved more defaults to metaflow_config

### PR DESCRIPTION
**Pull request by romain-intel**
Metadata providers are now plugins. We include the two that were already there
(local and service) but other metadata providers can now be easily added.

### CODE REVIEW

File: metaflow/metaflow_config.py [hunk 1]
```diff
@@ -41,9 +41,15 @@ def from_conf(name, default=None):
 ###
 # Default configuration
 ###
+DEFAULT_AUTH = from_conf('METAFLOW_DEFAULT_AUTH', 'aws')
 DEFAULT_DATASTORE = from_conf('METAFLOW_DEFAULT_DATASTORE', 'local')
+DEFAULT_DATA_PROVIDER = from_conf('METAFLOW_DEFAULT_DATAPROVIDER', '')
```
Comment (seeravikiran) [inline metaflow/metaflow_config.py:46]: Nit: DEFAULT_DATAPROVIDER so its also consistent with the `from_conf` invocation across other values.

File: metaflow/package.py [hunk 2]
```diff
@@ -16,13 +17,13 @@
     import io
     BytesIO = io.BytesIO
 
-DEFAULT_SUFFIXES = ['.py']
+LIST_DEFAULT_SUFFIXES = DEFAULT_SUFFIXES.split(',')
```
Comment (seeravikiran) [inline metaflow/package.py:20]: DEFAULT_SUFFIXES_LIST is probably better, no?

File: metaflow/metaflow_config.py (hunk 1 above)
Comment (romain-intel) [reply metaflow/metaflow_config.py:46]: Done.

File: metaflow/package.py [hunk 3]
```diff
@@ -16,13 +17,13 @@
     import io
     BytesIO = io.BytesIO
 
-DEFAULT_SUFFIXES = ['.py']
```
Comment (tuulos) [inline metaflow/package.py:19]: how is this change related to this PR?

File: metaflow/datastore/s3.py [hunk 4]
```diff
@@ -105,7 +105,7 @@ def get_latest_tasks(cls,
                 task_urls = [task.url for task in task_objs]
             urls = []
             for task_url in task_urls:
-                for attempt in range(5):
```
Comment (tuulos) [inline metaflow/datastore/s3.py:108]: `MAX_ATTEMPTS` seems to be 6 but we used to have `range(5)`. Let's just double check that the change is intentional

File: metaflow/package.py (hunk 3 above)
Comment (romain-intel) [reply metaflow/package.py:19]: I moved this to the configuration file.

**Comment by romain-intel** [comment]
Replaced by #331.
